### PR TITLE
Add erc20 genericTransfer to ReleaseGold

### LINKED ISSUE #8698: ReleaseGold should support generic ERC20 transfers (barring native and locked CELO)
### Expected Behavior

`ReleaseGold.transfer` accepts arbitrary ERC20 address for rescuing any accrued tokens. Special care should be taken to ensure this cannot be used as a native CELO escape valve.

### Current Behavior

Any ERC20 (ie [cEUR](https://github.com/celo-org/celo-monorepo/issues/7157)) can be permanently lost if sent to a ReleaseGold instance.

### PR DESCRIPTION
**Pull request by yorhodes**
### Tested

Unit tests

### Related issues

- Fixes #8698

### Backwards compatibility

Checked in CI

### CODE REVIEW

File: packages/protocol/contracts/governance/ReleaseGold.sol [hunk 1]
```diff
@@ -171,6 +171,22 @@ contract ReleaseGold is UsingRegistry, ReentrancyGuard, IReleaseGold, Initializa
...
+   * @notice Wrapper function for any ERC-20 transfer function.
+   * @dev Protects against celo/locked celo balance changes.
+   */
+  function genericTransfer(address erc20, address to, uint256 value)
+    external
+    onlyWhenInProperState
+  {
+    uint256 balanceBefore = address(this).balance +
+      getLockedGold().getTotalLockedGold(address(this));
+    IERC20(erc20).transfer(to, value);
```
Comment (m-chrzan) [inline packages/protocol/contracts/governance/ReleaseGold.sol:194]: Would technically a "flash self-loan" of non released CELO be still possible here? I.e. `to` is a contract that receives CELO, uses it to make a profit, then returns it back here. Or is the fact that `transfer` limits gas enough of a defense against this?

Either way, would just checking that `erc20` is not the registered GoldToken address be more explicit?

**Review by m-chrzan — COMMENTED** [review]
Would be great to see some unit tests.

File: packages/protocol/contracts/governance/ReleaseGold.sol (hunk 1 above)
Comment (yorhodes) [reply packages/protocol/contracts/governance/ReleaseGold.sol:194]: interesting question re: the flashloan behavior
I initially thought that a malicious erc20 could affect the goldtoken balances somehow but I realize now that's not really possible

Comment (yorhodes) [reply packages/protocol/contracts/governance/ReleaseGold.sol:194]: not sure how to defend against the flash loan behavior really...

Comment (m-chrzan) [reply packages/protocol/contracts/governance/ReleaseGold.sol:194]: Wouldn't checking against the GoldToken address be sufficient?

Comment (yorhodes) [reply packages/protocol/contracts/governance/ReleaseGold.sol:194]: oh, yes. flash loan of **unreleased CELO** is impossible with the check I added

File: packages/protocol/test/governance/voting/release_gold.ts [hunk 2]
```diff
@@ -227,6 +227,36 @@ contract('ReleaseGold', (accounts: string[]) => {
...
+
+    it('should transfer stable token from the release gold instance', async () => {
+      const startBalanceFrom = await mockStableToken.balanceOf(releaseGoldInstance.address)
+      const startBalanceTo = await mockStableToken.balanceOf(receiver)
+      await releaseGoldInstance.genericTransfer(mockStableToken.address, receiver, transferAmount, {
+        from: beneficiary,
+      })
+      const endBalanceFrom = await mockStableToken.balanceOf(releaseGoldInstance.address)
+      const endBalanceTo = await mockStableToken.balanceOf(receiver)
+      assertEqualBN(endBalanceFrom, startBalanceFrom.minus(transferAmount))
```
Comment (m-chrzan) [inline packages/protocol/test/governance/voting/release_gold.ts:323]: Can we make these checks with absolute values? I.e. remove `startBalance*` and check against constants. Will make the test a little cleaner and more explicit.

**Review by m-chrzan — APPROVED** [review]
Just one small suggestion.

**Pull request merged**